src/currencyRatesFileGen.js

```diff
@@ -0,0 +1,59 @@
+// prebidCurrencyRatesFileGenerator
+// loads currency rates from api.fixer.io and creates a JSON representation in S3
+// revision:
+var http = require('http');
+var AWS = require('aws-sdk');
+
+var fromCurrencies = ['USD','GBP'];
+var supportedCurrencies = 'AUD,BRL,CAD,CHF,CNY,CZK,DKK,EUR,GBP,HKD,HUF,IDR,ILS,INR,JPY,KRW,MXN,MYR,NOK,NZD,PHP,PLN,RUB,SEK,SGD,THB,TRY,USD,ZAR';
+
+var bucket = 'currency.prebid.org';
+var filename = 'latest.json';
+var debug = true;
+var expires = 24 * 3600 + 5; // when to expire for HTTP "Expires:" header (seconds)
+
+exports.handler = function(event, context) {
+    var results = [];
+    var countCompleted = 0;
+    for (let fromCurrency of fromCurrencies) {
+        let url = 'http://api.fixer.io/latest?base=' + fromCurrency + '&symbols=' + supportedCurrencies;
+        log('requesting: ' + url);
+        let req = http.get(url, (res) => {
```

Comment:
these should be separate functions passed in to be testable.

Reply:
For clarification, on separating the inline code into functions..

line #19 would become: getCurrencyUrl(fromCurrency, supportedCurrencies)

line #21 would become: getHttpRequest(url, ...) <- should the 'http' object be passed as an argument for easier unit test stubbing?

Reply:
right so the idea is just refactoring into smaller, testable functions. That sounds like a good start. Up to you if you want to pass `http` into the function for testing - the other alternative is to use a sinon stub/sandbox for it. Since this is a small project, sinon might be overkill.

Reply:
Thanks for the fast response, I'd like to use sinon but the code size does make it seem like overkill, so I'll probably just mock http manually